### PR TITLE
chore: portalloc cleanup

### PR DESCRIPTION
**Pull request by tvolk131**

### CODE REVIEW

File: utils/portalloc/src/data/dto.rs [hunk 1]
```diff
@@ -32,14 +36,63 @@ impl Default for RootData {
...
+                            port, "Could not use a port. Will try a different range"
+                        );
+                        base_port = port + 1;
+                        continue 'retry;
+                    }
+                    Ok(l) => l,
+                };
+            }
+
+            const ALLOCATION_TIME_SECS: u64 = 120;
```
Comment (dpc) [inline utils/portalloc/src/data/dto.rs:92]: If you're refactoring it (which is a good idea), I would move the stuff from this line downwards to a separate function, so the logic here can be one larger loop-y thing on immutable `&self` and then mutations live outside. Unless I'm missing something here that makes it necessary to remain mutable.

File: utils/portalloc/src/data/dto.rs [hunk 2]
```diff
@@ -37,6 +43,8 @@ impl Default for RootData {
 
 impl RootData {
     pub fn get_free_port_range(&mut self, range_size: u16) -> u16 {
+        self.reclaim(now_ts());
```
Comment (dpc) [inline utils/portalloc/src/data/dto.rs]: This now will not work with `&self`, but possibly can be moved out too.

**Review by dpc — COMMENTED** [review]
Comment, but LGTM

**Comment by dpc** [comment]
Would be good to wrap up and land. :) @tvolk131 DM for a snappy response when ready.

**Comment by tvolk131** [comment]
Sorry, been swamped the past few days - will get to this tonight!

File: utils/portalloc/src/data/dto.rs (hunk 1 above)
Comment (tvolk131) [reply utils/portalloc/src/data/dto.rs:92]: I moved this code into `RootData::insert`. But the loop here calls `RootData::reclaim` which requires `&mut self`.

File: utils/portalloc/src/data/dto.rs (hunk 2 above)
Comment (tvolk131) [reply utils/portalloc/src/data/dto.rs]: I'm not sure what you mean here - can you clarify?

**Review by dpc — APPROVED** [review]

**Pull request merged**